src/relays/BasicRelay.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2021 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.8.9;
+
+import "../WormholeGUID.sol";
+
+interface DaiJoinLike {
+    function dai() external view returns (TokenLike);
+    function exit(address, uint256) external;
+    function join(address, uint256) external;
+}
+
+interface TokenLike {
+    function balanceOf(address) external view returns (uint256);
+    function transfer(address, uint256) external returns (bool);
+}
+
+interface WormholeOracleAuthLike {
+    function requestMint(
+        WormholeGUID calldata wormholeGUID,
+        bytes calldata signatures,
+        uint256 maxFeePercentage
+    ) external returns (uint256 postFeeAmount);
+    function wormholeJoin() external view returns (WormholeJoinLike);
+}
+
+interface WormholeJoinLike {
+    function wormholes(bytes32 hashGUID) external view returns (bool, uint248);
+}
+
+// Relay messages automatically on the target domain
+// User provides gasFee which is paid to the msg.sender
+contract BasicRelay {
+
+    DaiJoinLike            public immutable daiJoin;
+    TokenLike              public immutable dai;
+    WormholeOracleAuthLike public immutable oracleAuth;
+    WormholeJoinLike       public immutable wormholeJoin;
+
+    constructor(address _oracleAuth, address _daiJoin) {
+        oracleAuth = WormholeOracleAuthLike(_oracleAuth);
+        daiJoin = DaiJoinLike(_daiJoin);
+        dai = daiJoin.dai();
+        wormholeJoin = oracleAuth.wormholeJoin();
+    }
+
+    function relay(
+        WormholeGUID calldata wormholeGUID,
+        bytes calldata signatures,
+        address receiver,
+        uint256 maxFeePercentage,
+        uint256 gasFee,
```

Comment:
I would add a comment here that it's in DAI

Reply:
done